ManusGloves/include/ManusGloveHelper.h

```diff
@@ -0,0 +1,377 @@
+/**
+ * @file ManusGloveHelper.h
+ * @authors Ehsan Ranjbari <ehsan.ranjbari@iit.it>
+ * @copyright 2024 AMI Lab - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2024
+ */
+
+#ifndef MANUS_GLOVE_HELPER_HPP
+#define MANUS_GLOVE_HELPER_HPP
+
+#include <Eigen/Dense>
+
+// std
+#include <array>
+#include <iostream>
+#include <memory>
+#include <stdint.h>
+#include <mutex>
+#include <vector>
+#include <chrono> // Used for haptic commands.
+#include <cmath> // Used for rounding float values.
+#include <iomanip> // Used for printing glove data, and converting glove IDs to strings.
+
+// YARP
+#include <yarp/os/LogStream.h>
+#include <yarp/sig/Vector.h>
+
+// Manus Glove
+#include <ManusGlove.h>
+
+/**
+ * GloveControlHelper is an helper class for controlling the glove.
+ */
+namespace manusGlove
+{
+    enum class ClientReturnCode : int;
+    enum class ClientState : int;
+    class ManusGloveHelper;
+    class ClientSkeleton;
+    class ClientSkeletonCollection;
+    const std::string LogPrefix = "manusGlove::ManusGloveHelper::";
+}
+
+// The current state of the client.
+enum class manusGlove::ClientState : int
+{
+    ClientState_PickingConnectionType = 0,
+    ClientState_LookingForHosts,
+    ClientState_NoHostsFound,
+    ClientState_PickingHost,
+    ClientState_ConnectingToCore,
+    ClientState_DisplayingData,
+    ClientState_Disconnected,
+
+    ClientState_MAX_CLIENT_STATE_SIZE
+};
+
+// Values that can be returned by this application.
+enum class manusGlove::ClientReturnCode : int
+{
+    ClientReturnCode_Success = 0,
+    ClientReturnCode_FailedPlatformSpecificInitialization,
+    ClientReturnCode_FailedToResizeWindow,
+    ClientReturnCode_FailedToInitialize,
+    ClientReturnCode_FailedToFindHosts,
+    ClientReturnCode_FailedToConnect,
+    ClientReturnCode_UnrecognizedStateEncountered,
+    ClientReturnCode_FailedToShutDownSDK,
+    ClientReturnCode_FailedPlatformSpecificShutdown,
+    ClientReturnCode_FailedToRestart,
+    ClientReturnCode_FailedWrongTimeToGetData,
+
+    ClientReturnCode_MAX_CLIENT_RETURN_CODE_SIZE
+};
+
+// Used to store the information about the final animated skeletons.
+class manusGlove::ClientSkeleton
+{
+public:
+    SkeletonInfo info;
+    std::vector<SkeletonNode> nodes;
+};
+
+// Used to store all the final animated skeletons received from Core.
+class manusGlove::ClientSkeletonCollection
+{
+public:
+    std::vector<ClientSkeleton> skeletons;
+};
+
+class manusGlove::ManusGloveHelper
```

Comment:
I have the impression that this code has been copied from somewhere else. If yes, from where? Can we import it instead of copying it?

Reply:
Well, I followed the Sense Glove Dk1 (https://github.com/robotology/wearables/tree/master/devices/HapticGlove) structure as an example plus the Manus example code (can be found in the SDK file).